flake8_quotes/__init__.py

```diff
@@ -119,32 +107,24 @@ def parse_options(cls, options):
             #   -> {good_single: ', good_multiline: ''', bad_single: ", bad_multiline: """}
             cls.config.update(cls.MULTILINE_QUOTES[options.multiline_quotes])
 
-    def get_file_contents(self):
-        if self.filename in ('stdin', '-', None):
-            return stdin_get_value().splitlines(True)
-        else:
-            return readlines(self.filename)
-
     def run(self):
-        file_contents = self.get_file_contents()
-
-        noqa_line_numbers = self.get_noqa_lines(file_contents)
-        errors = self.get_quotes_errors(file_contents)
-
-        for error in errors:
+        self._tokens = [Token(t)
```

Comment:
**Required change:** While I understand that we're trying to cache tokens for `run()`, it feels off that we're setting a class property outside of `__init__`. Additionally, this reduces testability by disallowing for passing `tokens` as a parameter to other methods

Requesting to make `tokens` a passable parameter to methods like `self.get_quotes_errors()`:

```python
tokens = [Token(t) ...]
for error in self.get_quotes_errors(tokens)
```